share/sysbuild/CMakeLists.txt

```diff
@@ -72,6 +72,18 @@ include(${BOARD_DIR}/sysbuild.cmake OPTIONAL)
 # This allows image specific sysbuild.cmake to be processed.
 list(LENGTH IMAGES images_length)
 while(NOT "${images_length}" EQUAL "${processed_length}")
+  # Check for any duplicate entries in image names to prevent an infinite loop
+  set(TMP_IMAGES ${IMAGES})
+  list(REMOVE_DUPLICATES TMP_IMAGES)
+  list(LENGTH TMP_IMAGES tmp_images_length)
+
+  if(NOT "${images_length}" EQUAL "${tmp_images_length}")
+    message(FATAL_ERROR "A duplicate image name was provided, image names must be unique. "
```

Comment:
Seems you prefer to keep the fatal error.

Usually I would say we can't check every variable in use, but I acknowledge this is special case.

Instead of multiple list calls, just fetch new length in dedicated var and then assign to the loop var after checking.
If the var stays unaltered even though the image and processed list where different in length then the image list must have a duplicated entry.

Here you can just do:
https://github.com/zephyrproject-rtos/zephyr/blob/9a63abc5c606bdb9582c4fe70262c101037e9cf0/share/sysbuild/CMakeLists.txt#L83-L84

```
  list(LENGTH IMAGES images_length)
  list(LENGTH images_sysbuild_processed processed_length_new)

  if("${processed_length_new}" EQUAL "${processed_length}")
    # Image length was different than processed length, but no new images are processed.
    message(FATAL_ERROR "A duplicate image name was provided, image names must be unique.")
  endif()
  set(processed_length ${processed_length_new})
```